tests/integration/targets/disk_info/tasks/main.yml

```
@@ -0,0 +1,40 @@
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Test code for the disk_info module
+# Copyright 2021 VMware, Inc
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+- name: "Installing the psutil module"
+  pip:
+    name: psutil
+    extra_args: "-c {{ remote_constraints }}"
+
+- name: "Checking the information from df command"
+  shell:  "df -h | grep '^/dev/' | awk '{ print $1 }'"
+  register: result_df
+
+- name: "Checking the all disk info"
+  community.general.disk_info:
+  register: result_all_disk
+
+- name: "Checking the specific disk info"
+  community.general.disk_info:
+    name: "{{ result_df.stdout_lines[0] }}"
+  register: result_specific_disk
+
+- name: "Obtaining all disk info check"
+  assert:
+    that:
+      - "{{ result_df.stdout_lines | unique | length }} == {{ result_all_disk['disk_info'].keys() | length }}"
```

```suggestion
      - "result_df.stdout_lines | unique | length == result_all_disk['disk_info'].keys() | length"
```
These lines are not templated. You have to write a jinja2 expression here.